wallet/src/account/utxo_selector/mod.rs

```diff
@@ -0,0 +1,774 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BinaryHeap;
+
+use common::{
+    chain::{
+        tokens::{OutputValue, TokenData},
+        TxOutput, UtxoOutPoint,
+    },
+    primitives::{signed_amount::SignedAmount, Amount},
+};
+use crypto::random::{make_pseudo_rng, Rng, SliceRandom};
+
+const TOTAL_TRIES: u32 = 100_000;
+
+/// A group of UTXOs paid to the same output script.
+/// This helps reduce privacy leaks resulting from address reuse.
+#[derive(Clone)]
+pub struct OutputGroup {
+    /// The list of UTXOs contained in this output group.
+    outputs: Vec<(UtxoOutPoint, TxOutput)>,
+    /// the total amount of the outputs in this group
+    value: Amount,
+    /// The fee to spend these UTXOs at the effective feerate.
+    fee: Amount,
+    /// The fee to spend these UTXOs at the long term feerate.
```

Comment:
The English here is a little broken "The fee to spend these UTXOs"... I'm not sure I understand what this means.

Same applies for the long_term_fee below.

Reply:
it refers to the amount of fee that we will need to pay if we include these Utxos in the transaction based on the size in bytes they will take up.
The normal fee should be based on the current fee rate (maybe there is a spike or dip in fee), and the long-term fee is the estimated fee rate overall.

Reply:
yes

Reply:
I would then make the comments "The fee cost of these UTXOs..."